### PR TITLE
libphonenumber: init at 8.9.9

### PR DESCRIPTION
**Pull request by illegalprime**
###### Motivation for this change

Adds `libphonenumber` a handy library to parse and work with phone number of different formats.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This is my first package contribution here, so I'm sorry in advance if I don't have everything formatted.

### CODE REVIEW

**Comment by FRidh** [comment]
Thanks. Please do check https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes.

**Comment by FRidh** [comment]
@GrahamcOfBorg build libphonenumber

**Comment by FRidh** [comment]
@GrahamcOfBorg build phonenumber

**Comment by illegalprime** [comment]
Thanks! Who should I put as the maintainer?

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    protobuf
+    icu
+  ];
+
+  meta = {
+    description = "Google's i18n library for parsing and using phone numbers";
+    license = stdenv.lib.licenses.asl20;
+  };
+
+  configurePhase = "cmake -DCMAKE_INSTALL_PREFIX=$out cpp";
```
Comment (matthewbauer) [inline pkgs/development/libraries/libphonenumber/default.nix]: I think you should be able to leave this out & move it to cmakeFlags.

**Comment by illegalprime** [comment]
@matthewbauer how would I specify that the directory with the `CMakeLists.txt` is `cpp`?

**Comment by matthewbauer** [comment]
You should just be able to put it in cmakeFlags I believe. It’s not that important though sice i think what you have is equivalent.

You should mark the package as linux only here though because it looks like there is some issues on macOS.

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  version = "8.9.9";
+in
+stdenv.mkDerivation {
+  name = "phonenumber-${version}";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://github.com/googlei18n/libphonenumber/archive/v${version}.tar.gz";
+    sha256 = "1m2z5hk2yg99sqi70wf26biprhppklwxwyk0g717c1x6yybch13g";
+  };
```
Comment (Infinisil) [inline pkgs/development/libraries/libphonenumber/default.nix]: Use `fetchFromGitHub` instead.

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  nativeBuildInputs = [
+    cmake
+  ];
```
Comment (Infinisil) [inline pkgs/development/libraries/libphonenumber/default.nix]: `pkgconfig` and `gmock` should probably be in `nativeBuildInputs`.

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    pkgconfig
+    protobuf
+    icu
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Google's i18n library for parsing and using phone numbers";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ illegalprime ];
+  };
```
Comment (Infinisil) [inline pkgs/development/libraries/libphonenumber/default.nix]: Nitpick: Can you move this meta section to the very bottom? It's somewhat of a convention

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+  configurePhase = "cmake -DCMAKE_INSTALL_PREFIX=$out cpp";
+
+  checkPhase = "./libphonenumber_test";
+
+  doCheck = true;
```
Comment (Infinisil) [inline pkgs/development/libraries/libphonenumber/default.nix]: I think `doCheck` is true by default if I'm not mistaken.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -4470,6 +4470,8 @@ with pkgs;
 
   phodav = callPackage ../tools/networking/phodav { };
 
+  phonenumber = callPackage ../development/libraries/libphonenumber { };
```
Comment (Infinisil) [inline pkgs/top-level/all-packages.nix]: Use `libphonenumber` instead, and put it in the section of `all-packages.nix` where other libraries are.

File: pkgs/development/libraries/libphonenumber/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    icu
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Google's i18n library for parsing and using phone numbers";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ illegalprime ];
+  };
+
+  configurePhase = "cmake -DCMAKE_INSTALL_PREFIX=$out cpp";
```
Comment (Infinisil) [inline pkgs/development/libraries/libphonenumber/default.nix]: Remove this and use `cmakeDir = "../cpp";` instead.

**Review by infinisil — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (illegalprime) [reply pkgs/top-level/all-packages.nix]: I thought this was alphabetical?

**Comment by illegalprime** [comment]
Is it better to have `version` as an argument to this function? Like `{ version ? "8.9.9" }`? Then it would be easy to build different versions, or my understanding of this is just off.

**Comment by Infinisil** [comment]
Nah, Nix doesn't generally handle versions like that. And overriding it like that wouldn't change anything in fact, because the source will still point to the old version. This PR looks good to me like this, thanks :)

**Review by infinisil — APPROVED** [review]

**Comment by Infinisil** [comment]
Oh wait one more thing: Can you make a first commit for adding you as a maintainer and a second one for adding the package?

**Comment by Infinisil** [comment]
Ping

**Comment by illegalprime** [comment]
Moving across cities now, I'll try to get to this tonight. Thanks for the
reminder!

**Pull request merged**